test/ci-scripts/Run-TemplateAnalyzer.ps1

```diff
@@ -56,21 +56,22 @@ function Analyze-Template {
     }
 }
 
-$reportedErrors = $false
-Get-ChildItem $sampleFolder -Directory | # To analyze all the JSON files in folders that could contain nested templates
+$passed = $true
+$preReqsFolder = "$sampleFolder\prereqs"
+Get-ChildItem $sampleFolder -Recurse -Filter *.json |
     ForEach-Object {
-        if ($_.Name -ne "prereqs") {
-            Get-ChildItem $_ -Recurse -Filter *.json |
-                ForEach-Object {
-                    $reportedErrors = $reportedErrors -or (Analyze-Template $_.FullName)
-                }
+        $params = @{ "templateFilePath" = $_.FullName }
+        if ($_.FullName -eq "$preReqsFolder\$prereqTemplateFilename") {
+            $params.Add("parametersFilePath", "$preReqsFolder\$prereqParametersFilename")
+        } elseif ($_.FullName -eq "$sampleFolder\$mainTemplateFilename") {
+            $params.Add("parametersFilePath", "$sampleFolder\$mainParametersFilename")
         }
+
+        $newAnalysisPassed = Analyze-Template @params
+        $passed = $passed -and $newAnalysisPassed # evaluation done in two lines to avoid PowerShell's lazy evaluation
     }
-$preReqsFolder = "$sampleFolder\prereqs"
-$reportedErrors = $reportedErrors -or (Analyze-Template "$preReqsFolder\$prereqTemplateFilename" "$preReqsFolder\$prereqParametersFilename")
-$reportedErrors = $reportedErrors -or (Analyze-Template "$sampleFolder\$mainTemplateFilename" "$sampleFolder\$mainParametersFilename")
 
-Write-Host "##vso[task.setvariable variable=templateAnalyzer.reportedErrors]$reportedErrors"
-Write-Host "##vso[task.setvariable variable=templateAnalyzer.output.filePath]$testOutputFilePath"
+Write-Host "##vso[task.setvariable variable=template.analyzer.result]$passed"
+Write-Host "##vso[task.setvariable variable=template.analyzer.output.filePath]$testOutputFilePath"
```

Comment:
Do we need to write this var or can it be a static pipeline var?  Seems like it never changes so could be a param pulled from the pipeline?

Reply:
It depends on $ttk.folder, I didn't want to assume that one doesn't change, or that we will remember to update this one if ttk.folder changes

Reply:
we can reference the ttk.folder (one var can reference another)... though TBH I don't think it matters where you put the file.  we would define the var in either case, since it's easier to keep track of them across tasks that way...